tools/generator/Java.Interop.Tools.Generator.Transformation/EnumMappings.cs

```diff
@@ -139,85 +141,36 @@ internal List<string> WriteEnumerations (string output_dir, Dictionary<string, E
-			foreach (var enu in enums) {
-				var path    = Path.Combine (output_dir, GetFileName (enu.Key, useShortFileNames) + ".cs");
-				files.Add (path);
-				using (StreamWriter sw = new StreamWriter (path, append: false)) {
-					string ns = enu.Key.Substring (0, enu.Key.LastIndexOf ('.')).Trim ();
-					string enoom = enu.Key.Substring (enu.Key.LastIndexOf ('.') + 1).Trim ();
-
-					sw.WriteLine ("namespace {0} {{", ns);
-					if (enu.Value.BitField)
-						sw.WriteLine ("  [System.Flags]");
-					sw.WriteLine ("  public enum {0} {{", enoom);
-
-					foreach (var member in enu.Value.Members) {
-						var managedMember = FindManagedMember (ns, enoom, enu.Value, member.Key, gens);
-						sw.WriteLine ("    [global::Android.Runtime.IntDefinition (" + (managedMember != null? "\"" + managedMember + "\"" : "null") + ", JniField = \"" + StripExtraInterfaceSpec (enu.Value.JniNames [member.Key]) + "\")]");
-						sw.WriteLine ("    {0} = {1},", member.Key.Trim (), member.Value.Trim ());
-					}
-					sw.WriteLine ("  }");
-					sw.WriteLine ("}");
-				}
-			}
-			return files;
-		}
-
-		string StripExtraInterfaceSpec (string jniFieldSpec)
-		{
-			return jniFieldSpec.StartsWith ("I:", StringComparison.Ordinal) ? jniFieldSpec.Substring (2) : jniFieldSpec;
-		}
-
-		string FindManagedMember (string ns, string enumName, EnumDescription desc, string enumFieldName, IEnumerable<GenBase> gens)
-		{
-			if (desc.FieldsRemoved)
-				return null;
+			var files = new ConcurrentBag<string> ();
 
-			var jniMember = desc.JniNames [enumFieldName];
-			if (string.IsNullOrWhiteSpace (jniMember)) {
-				// enum values like "None" falls here.
-				return null;
-			}
-			return FindManagedMember (jniMember, gens);
-		}
-
-		WeakReference cache_found_class;
+			Parallel.ForEach (enums, enu => {
+				var path = Path.Combine (output_dir, GetFileName (enu.Key, useShortFileNames) + ".cs");
+				files.Add (path);
 
-		string FindManagedMember (string jniMember, IEnumerable<GenBase> gens)
-		{
-			string package, type, member;
-			ParseJniMember (jniMember, out package, out type, out member);
-			var fullJavaType = (string.IsNullOrEmpty (package) ? "" : package + ".") + type;
-
-			var cls = cache_found_class != null ? cache_found_class.Target as GenBase : null;
-			if (cls == null || cls.JniName != fullJavaType) {
-				cls = gens.FirstOrDefault (g => g.JavaName == fullJavaType);
-				if (cls == null) {
-					// The class was not found e.g. removed by metadata fixup.
-					return null;
+				using (var sw = File.CreateText (path)) {
+					var generator = new EnumGenerator (sw);
+					generator.WriteEnumeration (enu, gens);
 				}
-				cache_found_class = new WeakReference (cls);
-			}
-			var fld = cls.Fields.FirstOrDefault (f => f.JavaName == member);
-			if (fld == null) {
-				// The field was not found e.g. removed by metadata fixup.
-                                return null;
-			}
-			return cls.FullName + "." + fld.Name;
+			});
+
+			return files.ToList ();
```

Comment:
I believe that this should be `files.OrderBy (f => f).ToList ()`, or some variation thereof.

Rationale: [C# deterministic builds](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-options/deterministic-compiler-option).  If, for whatever reason, `generator` is re-executed during a build, it would be nice if, when deterministic builds are enabled, the resulting assembly *did not change* unless the generated code actually changed.

The order of generated files appears to be used in the `csc /deterministic` algorithm, and thus altering the order of files may result in the creation of a different assembly.  For example:

```csharp
// app1.cs
class App1 {
	public static void Main ()
	{
		App2 a = new App2 ();
	}
}
```

```csharp
// app2.cs
class App2 {
}
```

```bash
$ csc /deterministic /out:upwards.exe app1.cs app2.cs
$ csc /deterministic /out:downwards.exe app2.cs app1.cs
$ md5 downwards.exe upwards.exe 
MD5 (downwards.exe) = cb7690e4878515303bde517447130b28
MD5 (upwards.exe) = b258bd49fe38709db5191f1adef9afc9
```

"Simply" changing the order of command-line arguments results in a different assembly.

Reply:
We do the sort when we write the project file:
https://github.com/xamarin/java.interop/blob/master/tools/generator/GenerationInfo.cs#L64

Reply:
Ah, I missed that.  Thanks!